pkg/kube/interface.go

```diff
@@ -68,6 +70,12 @@ type Interface interface {
 
 	// IsReachable checks whether the client is able to connect to the cluster.
 	IsReachable() error
+
+	// GetPodList list all pods that match the specified listOptions
+	GetPodList(namespace string, listOptions metav1.ListOptions) (*v1.PodList, error)
```

Comment:
Adding new methods to an interface is considered a breaking change. So this PR would either need to refactor this or defer it to Helm 4.

Reply:
Just to be clearer -- I was looking at the above while getting ready for the discussion on the Helm call. Sorry for the late comment. But, yes, adding methods to a public interface is considered a breaking change, and thus has to be deferred to a major release. However, I think this one should be easy enough to refactor into a separate interface and some assertions about interface type. (The Go codebase itself does this quite a bit).

Reply:
Hi @technosophos - I have refactored the new methods on to the `InterfaceExt` interface similarly to `WaitForDelete`. I'd hope this is sufficient and we can move all breaking change methods to `Interface` as part of Helm 4. I have added a TODO next to the type assertions as they should be able to be removed also once methods are moved to Interface. Please let me know if any issues, thanks again!

Reply:
Thanks for the refactor - we should be able to consider this now.

Reply:
Thanks very much @bridgetkromhout !

Reply:
I'm sorry to say that the Go guidelines say that extending ANY interface is a breaking change. You need to create a new Interface for this. Something like `InterfaceLogs`. The Go article that we link in the Helm backwards compatibility rules is at https://go.dev/blog/module-compatibility#working-with-interfaces. This describes what to do.

Reply:
Hi Matt, apologies i've just seen this. I thought the entire point of `InterfaceExt` was to act as a temporary extension until the changes could be merged in to `Interface` in helm4 from the doc string:

```golang
// InterfaceExt is introduced to avoid breaking backwards compatibility for Interface implementers.
//
// TODO Helm 4: Remove InterfaceExt and integrate its method(s) into the Interface.
```